File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int s=0;
        int p=1;
        int temp=n;
        while (temp>0){
            int r;
            r=temp%10;
            s+=r;
            p*=r;
            temp/=10;
        }
        if(n%(s+p)==0) return true;
      return false;  
    }
};

File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int x = n;
        int sum = 0;
        int p = 1;

        while (n > 0) {
            int r = n % 10;
            sum += r;
            p *= r;
            n /= 10;
        }

        int rev = sum + p;
        return x % rev == 0;
    }
}